### DISCUSSION TITLE
2021/01/10/managing-python-on-macos-big-sur/

### DISCUSSION THREAD

**Discussion by giscus** [russmckendrick/blog#8] [Comment]
# 2021/01/10/managing-python-on-macos-big-sur/

When Apple releases a new version of macOS it always takes a few months for everything to catchup, following my last blog post where I mentioned that I was having problems installing Python on macOS Big Surwhich meant that my preferred method of installing and managing Python using pyenv, which is documented in this blog post, didn’t work out of the box — and the workarounds suggested workarounds on GitHub made my shell really slow.

https://www.russ.cloud/2021/01/10/managing-python-on-macos-big-sur/

**Comment by andypb** [comment]
Thanks for this, very useful. To get it to work after installing pyenv with Homebrew I had to only write this to .zshrc, i.e. omitting the export environment variable commands:
`echo -e $'if command -v pyenv 1>/dev/null 2>&1; then\\n eval "$(pyenv init --path)"\\n  eval "$(pyenv init -)"\\nfi' >> ~/.zshrc`
